packages/contracts-bedrock/src/L1/OptimismPortal.sol

```diff
@@ -160,6 +163,7 @@ contract OptimismPortal is Initializable, ResourceMetering, ISemver {
         if (l2Sender == address(0)) {
             l2Sender = Constants.DEFAULT_L2_SENDER;
         }
+        _balance = _initialBalance;
```

Comment:
We could change it to 

_balance = balance();

and remove _initialBalance from function signature. (It is smaller change and possibly easier to upstream it later on).

Reply:
I had to revert those changes again, since it introduced a circular dependency in the initialization order.
Putting my findings here for reference:

The `SystemConfig` sets the gas-paying token on the `OptimismPortal.setGasPayingToken`  during initialization. This seems to emit an event that is used in L1 derivation. And our changes let the `SystemConfig` call the aforementioned `SystemConfig.gasPayingToken()` during it's initialization.

Reply:
We use [this](https://github.com/ethereum-optimism/optimism/blob/e00f23ad0208f2e35aef5435d8a3d2e369144419/packages/contracts-bedrock/src/universal/StorageSetter.sol#L4) contract to set storage during upgrades by swapping the implementation to this contract and then to the desired implementation atomically. This prevents complexity around coupling storage migrations with implementation code and would allow this diff to go away. This pattern will be required for contract upgrades since we need to zero out the `_initialized` slot

Reply:
Hey @tynes - thank you for reaching out. This is a great idea, we will add this to our next iteration concerning the custom gas token feature, as we were not happy with modifying Optimism core contracts in the first place.